fix(client): Vague Error messages

thecodingaviator opened the pull request:
<!-- Please follow this checklist and put an x in each of the boxes, like this: [x]. It will ensure that our team takes your pull request seriously. -->

- [x] I have read [freeCodeCamp's contribution guidelines](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/CONTRIBUTING.md).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `master` branch of freeCodeCamp.
- [x] None of my changes are plagiarized from another source without proper attribution.
~- [x] All the files I changed are in the same world language (for example: only English changes, or only Chinese changes, etc.)~
- [x] My changes do not use shortened URLs or affiliate links.

<!--If your pull request closes a GitHub issue, replace the XXXXX below with the issue number.-->

Closes #17413

thecodingaviator committed:
fix(client): Vague Error messages

raisedadead commented on api-server/common/models/User-Identity.js:
```diff
@@ -50,9 +50,9 @@ export default function(UserIdent) {
-    Oops... something is not right. We did not find a valid email from your
-    ${provider} account. Please try again with a different provider that has an
-    email available with it.
+          ${provider} did not return a valid email address.
```
Fix the indentation like it was originally.

raisedadead commented on api-server/common/models/User-Identity.js:
```diff
@@ -50,9 +50,9 @@ export default function(UserIdent) {
-    ${provider} account. Please try again with a different provider that has an
-    email available with it.
+          ${provider} did not return a valid email address.
+          Please try again with a different account that has an
```
Fix the indentation like it was originally.

raisedadead commented on api-server/common/models/User-Identity.js:
```diff
@@ -50,9 +50,9 @@ export default function(UserIdent) {
-    email available with it.
+          ${provider} did not return a valid email address.
+          Please try again with a different account that has an
+    email associated with it or modify permissions to allow us to view your email.
```
```suggestion
    email associated with it or update the settings on ${provider}, for us to be able to retrieve your email.
```

thecodingaviator replied:
How about `update your settings on ${provider}`?

raisedadead commented on api-server/server/boot/authentication.js:
```diff
@@ -53,7 +53,7 @@ module.exports = function enableAuthentication(app) {
         throw wrapHandledError(new Error('could not destroy session'), {
           type: 'info',
-          message: 'Oops, something is not right.',
+          message: 'We could not log you out, please try logging out again.',
```
```suggestion
          message: 'We could not log you out, please try again in a moment.',
```

raisedadead replied:
Give them the impression, that something is delayed on our side.

raisedadead commented on api-server/server/boot/user.js:
```diff
@@ -202,10 +202,15 @@ function createPostReportUserProfile(app) {
+    } else if (!report || report === '') {
+      return res.json({
+        type: 'danger',
+        message: 'Your report cannot be empty. Please add information.'
```
```suggestion
        message: 'Your report cannot be empty. Please add details for us to be able to address your concern.'
```

raisedadead commented on api-server/server/boot/user.js:
```diff
@@ -202,10 +202,15 @@ function createPostReportUserProfile(app) {
-        message: 'Oops, something is not right please re-check your submission.'
+        message: 'Please enter a valid username.'
+      });
+    } else if (!report || report === '') {
```
This conditional is not valid. What happens when `username`, `report` both are null/undefined?

raisedadead commented on api-server/server/middlewares/error-handlers.js:
```diff
@@ -26,7 +26,7 @@ export default function prodErrorHandler() {
     const redirectTo = handled.redirectTo || `${homeLocation}/`;
     const message =
-      handled.message || 'Oops! Something went wrong. Please try again later';
+      handled.message || 'Oops! Something went wrong. Please try again';
```
```suggestion
      handled.message || 'Oops! Something went wrong. Please try again in a moment.';
```

thecodingaviator committed:
Apply suggestions from code review

Minor rewordings

Co-Authored-By: mrugesh <1884376+raisedadead@users.noreply.github.com>

thecodingaviator committed:
fix indentation and change checksn

Added extra checks for when everything is null

try to fix pretties errors

--no-message

raisedadead reviewed (changes requested):
This entire block is anti-pattern. Why? Repetitive, Complex and difficult to read.

```js
    if (!username && (!report || report === '')) {
      return res.json({
      return res.json({
        type: 'danger',
        type: 'danger',
        message: 'Oops, something is not right please re-check your submission.'
        message: 'Please enter a valid username and add details for us to be able to address your concern.'
      });
    } else if ((!report || report === '') && username) {
      return res.json({
        type: 'danger',
        message: 'Your report cannot be empty. Please add details for us to be able to address your concern.'
      });
    } else if (!username) {
      return res.json({
        type: 'danger',
        message: 'Please enter a valid username.'
      });
      });
    }
    }
```
You should re-factor this to be a simple return early pattern.

raisedadead commented:
Finally I have noticed that you depend on the CI for while fixing up PRs, IMHO you should test things locally before pushing to ease the notifications barrage it triggers for everyone subscribed to the thread.

thecodingaviator committed:
revert changes to user.js

thecodingaviator committed:
remove dedent from error-handlers

raisedadead commented:
You should probably have different returns for each of those last validations on username and report, returning early. Anyways I'll merge this.
